### ISSUE TITLE
Interpolation of airfoil tables in aerodyn

### ISSUE THREAD

**Issue by tonyinme** [OpenFAST/openfast#374]
**Is your feature request related to a problem? Please describe.**
The lift and drag coefficient tables inside of aerodyn are discontinuous. When computing blade forces, the airfoil sections change abruptly from on point to the next.
This leads to discontinuities in the forces along the blade.

**Describe the solution you'd like**
There should be a variable for the total number of aerodyn points to be used.
Then, all values needed for the points can be interpolated from the tables.
The best solution would be to interpolate the lift and drag coefficients between sections at runtime.
This interpolation should be linear based on radial location.

**Describe alternatives you've considered**
As a work-around, one can define many blade sections each with it's own table of interpolated lift and drag coefficients.

**Additional context**
This feature is of great interest when running actuator line simulations in Nalu-Wind/SOWFA.

**Comment by andrew-platt** [comment]
To clarify, are you requesting interpolation of airfoil tables between spanwise locations?

**Comment by jjonkman** [comment]
Hi @tonyinme,

We had previously considered adding the ability in AeroDyn to interpolate airfoil data between spanwise stations, but chose not to for several reasons.  For one, the interpolation would not be straightforward considering the nonlinearity of the airfoil data.  For example, if one airfoil has the stall point around 10 degrees, and the second airfoil has a stall point around 20 degrees, how should this be interpolated for an airfoil section in between?  A simple linear interpolation at each angle of attack would result in two stall points (one at 10 and another at 20), which is not physical.  It is more intuitive for the stall point at the midpoint between the two stations to be at 15 degrees, but what interpolation would do this?  Second, how would one interpolate the unsteady airfoil aerodynamic parameters, when these should be based on the airfoil data themselves?

In the end, we decided that the aerodynamic solution is strongly tied to the accuracy of the airfoil tables, so, the user should be heavily involved in their specification.  Thus, we chose to force the user to specify the airfoil data at each and every station, and it is up to the user to ensure that the airfoil data is smooth between stations if that is desired for the response.  We have made available the AirfoilPrep preprocessor (https://nwtc.nrel.gov/AirFoilPrep) to aid in the specification of the airfoil data,  though AirfoilPrep is due for an update.

Best regards,

**Comment by tonyinme** [comment]
hey @andrew-platt, yes, this is a request for interpolation of airfoil tables between sections. As a part of this, it seems natural to interpolate all quantities to as many sections as needed.

@jjonkman I see your point. You think we can reconsider that decision? I still believe it would be beneficial to interpolate in between sections, even with unsteady effects. In terms of usability and running simulations with the actuator line model, this would simplify a lot of things. I know DTU interpolates based on thickness value, maybe this would be even better than the linear interpolation based on radial location?

**Comment by jjonkman** [comment]
@tonyinme,

Yes, interpolating based on thickness implies that one is interpolating within the same airfoil family, which are likely have much more similar airfoil data (e.g., the same stall point) than two possibly completely different airfoils.  AeroDyn already supports interpolation based on Reynolds number (or control) for this reason--see the recently merged in pull request #213.

**Comment by s1291** [comment]
Dear @jjonkman ,
Do you have any updates on this? Also, would you recommend any tools for this purpose?

**Comment by jjonkman** [comment]
Dear @s1291,

Airfoil data interpolation has not been added within AeroDyn.

However, airfoil data preprocessing tools such as the older Excel-based tool AirfoilPrep (https://www.nlr.gov/wind/nwtc/airfoil-prep) and the newer Python-based tool AirfoilPreppy (https://github.com/NLRWindSystems/AirfoilPreppy) do include functionality to interpolate between two sets of airfoil data (called "blending"), e.g., based on thickness within an airfoil family.

Best regards,
